theano/ifelse.py

```diff
@@ -245,12 +246,14 @@ def thunk():
                     else:
                         for out, outtype, t in izip(outputs, outtypes, ts):
                             compute_map[out][0] = 1
+                            val = storage_map[t][0]
                             if self.as_view:
-                                oval = outtype.filter(storage_map[t][0])
+                                storage_map[out][0] = val
+                            # Work around broken numpy deepcopy
+                            elif type(a) in (numpy.ndarray, numpy.memmap):
```

Comment:
Also, can you simply use `_less_broken_deepcopy`? I think it is essentially doing the same thing.